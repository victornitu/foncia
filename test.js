const axios = require('axios')

const baseUrl = 'http://localhost:8080'

const testEndpoint = (url, code, name) => {
  test(name, async () => {
    expect.assertions(2)
    try {
      const { status, data } = await axios.get(`${baseUrl}/${url}`)
      expect(status).toEqual(code)
      expect(data).toMatchSnapshot()
    } catch (e) {
      console.error(e)
    }
  })
}

const testError = (url, code, name) => {
  test(name, async () => {
    expect.assertions(2)
    try {
      await axios.get(`${baseUrl}/${url}`)
    } catch ({ response }) {
      const { status, data } = response
      expect(status).toEqual(code)
      expect(data).toMatchSnapshot()
    }
  })
}

testEndpoint('', 200, 'API is running')
testEndpoint('client', 200, 'API returns a list of clients')
testEndpoint('lot', 200, 'API returns a list of lots')
testError('unknown', 404, 'API returns a not found error')

test('API creates a client and a lot', async () => {
  expect.assertions(6)
  const client = {
    fullname: 'Victor Nitu',
    email: 'nitu.v.r@gmail.com',
    email2: 'victor.nitu@vvfluxembourg.com',
    telDomicile: '+33 123 123',
    telPro: '+44 234 234',
    telMobile: '+32 345 345',
    telMobile2: '+352 456 456',
    fax: '+49 567 567',
    sexe: 'M'
  }
  const lot = {
    surface: 35
  }
  try {
    // Create a new client
    const clientResponse = await axios.post(`${baseUrl}/client`, client)
    expect(clientResponse.status).toEqual(201)
    expect(clientResponse.data).toMatchObject(client)
    // Check new client is in the list of clients
    const clientId = clientResponse.data._id
    const clients = await axios.get(`${baseUrl}/client?page=1&size=999`)
    const savedClient = clients.data.clients.find(c => c._id === clientId)
    expect(savedClient).toEqual(clientResponse.data)
    // Set lot client to the newly created client
    lot.client = clientId
    // Create a new lot
    const lotResponse = await axios.post(`${baseUrl}/lot`, lot)
    expect(lotResponse.status).toEqual(201)
    expect(lotResponse.data).toMatchObject(lot)
    // Check new lot is in the list of lots
    const lotId = lotResponse.data._id
    const lots = await axios.get(`${baseUrl}/lot?page=1&size=999`)
    const savedLot = lots.data.lots.find(l => l._id === lotId)
    expect(savedLot).toEqual(lotResponse.data)
  } catch (e) {
    console.error(e)
  }
})
