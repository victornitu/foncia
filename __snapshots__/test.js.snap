// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API is running 1`] = `
Object {
  "status": "running",
}
`;

exports[`API returns a list of clients 1`] = `
Object {
  "clients": Array [
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae2f6",
      "email": "Rowland_Adams@yahoo.com",
      "email2": "Barbara.Ernser29@hotmail.com",
      "fax": "565-128-4346",
      "fullname": "Mrs. Craig Wehner",
      "sexe": "M",
      "telDomicile": "359.379.4026",
      "telMobile": "425-241-3599 x221",
      "telMobile2": "206.017.6574 x234",
      "telPro": "026.837.9868 x468",
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae2f7",
      "email": "Mckayla.Bashirian58@yahoo.com",
      "email2": "Anissa_Rowe22@gmail.com",
      "fax": "811-173-7208 x1000",
      "fullname": "Evert Sanford",
      "sexe": "F",
      "telDomicile": "(635) 020-9399 x3473",
      "telMobile": "(061) 205-8187",
      "telMobile2": "952.035.8808 x17687",
      "telPro": "(496) 529-8637",
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae2f8",
      "email": "Molly_Heidenreich@yahoo.com",
      "email2": "Christelle71@yahoo.com",
      "fax": "1-910-616-1370 x9932",
      "fullname": "Kody Grimes",
      "sexe": "M",
      "telDomicile": "603.434.5348",
      "telMobile": "1-090-514-3977",
      "telMobile2": "(278) 567-4818 x68456",
      "telPro": "649-323-3798",
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae2f9",
      "email": "Aric.Tremblay0@hotmail.com",
      "email2": "Ova.Jenkins69@yahoo.com",
      "fax": "916.049.6188 x625",
      "fullname": "Cecilia Mante",
      "sexe": "M",
      "telDomicile": "(217) 271-5777 x25165",
      "telMobile": "1-673-377-2839",
      "telMobile2": "(301) 594-2456 x3986",
      "telPro": "(738) 199-5414",
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae2fa",
      "email": "Wilfredo62@yahoo.com",
      "email2": "Francisca70@yahoo.com",
      "fax": "1-644-572-3024",
      "fullname": "Ms. Ian Crooks",
      "sexe": "M",
      "telDomicile": "(715) 225-1415",
      "telMobile": "106-076-5430",
      "telMobile2": "1-865-811-7791",
      "telPro": "338-169-3927 x28237",
    },
  ],
  "next": "?page=2&size=5",
  "previous": null,
}
`;

exports[`API returns a list of lots 1`] = `
Object {
  "lots": Array [
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae370",
      "client": "5be0e99d9228d14c03dae2f6",
      "surface": 35,
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae371",
      "client": "5be0e99d9228d14c03dae2f6",
      "surface": 27,
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae372",
      "client": "5be0e99d9228d14c03dae2f7",
      "surface": 25,
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae373",
      "client": "5be0e99d9228d14c03dae2f7",
      "surface": 26,
    },
    Object {
      "__v": 0,
      "_id": "5be0e99d9228d14c03dae374",
      "client": "5be0e99d9228d14c03dae2f7",
      "surface": 36,
    },
  ],
  "next": "?page=2&size=5",
  "previous": null,
}
`;

exports[`API returns a not found error 1`] = `
Object {
  "error": "endpoint not found",
}
`;
